mautic/mautic
# DPMMA-2524 fix message queue errors

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ Y ]
| New feature/enhancement? (use the a.x branch)      | [ N ]
| Deprecations?                          | [ N ]
| BC breaks? (use the c.x branch)        | [ N ]
| Automated tests included?              | [ Y ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #13070 #13060



#### Description:
This PR fixes two issues with the message queue:
#13060 - The `$this->em->detach($event);` should be called only if the event exists
#13070 - The companies should not be parsed like the custom field in the `setFields` method because it results in some weird value normalization:
![image](https://github.com/mautic/mautic/assets/8580942/5f36f231-9d15-466f-8790-dc6fb4179255)



#### Steps to test this PR:
1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Set the Frequency rule in the Mautic settings to 1 message per day.
3. Create a segment with some contacts, one contact should have an empty first name (null in the database) and at least one company.
4. Create and send an email to the segment, add the `{contactfield=firstname}` and {contactfield=companyname} to the email content to ensure that the tokens work correctly after changes
5. Clone the email and send it again - it should be added to the message queue because of the limit.
6. After 24 hours, run the command to send the queue `ddev exec php bin/console mautic:messages:send`.
